validationparser/interceptor.go

```diff
@@ -0,0 +1,57 @@
+package validator
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	"github.com/infobloxopen/atlas-app-toolkit/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+type validator interface {
+	Validate() error
+}
+
+// UnaryServerInterceptor returns a new unary server interceptor that validates incoming messages.
+//
+// Invalid messages will be rejected with `InvalidArgument` before reaching any userspace handlers.
+func UnaryServerInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		if v, ok := req.(validator); ok {
+			if err := v.Validate(); err != nil {
+				return nil, err
+			}
+		}
+		return handler(ctx, req)
+	}
+}
+
+// MapValidationError function returns a MapFunc that checks if the error is of type validation
+func MapValidationError() errors.MapFunc {
```

Comment:
I suggest to call it NewValidationErrorMapping()

Reply:
I updated it to ValidationErrorMapping because adding "New" might be a little confusing.